doozerlib/cli/scan_sources.py

```diff
@@ -94,21 +103,28 @@ def add_assessment_reason(meta, changing, reason):
                     else:
                         runtime.logger.warning(f"Appears that {dgk} has never been built before; can't determine package name")
             elif meta.meta_type == 'image':
-                if reason:
-                    add_assessment_reason(meta, needs_rebuild, reason=reason)
+                if not needs_rebuild:  # If no change has been detected, check configurations like image meta, repos, and streams to see if they have changed
+                    # We detect config changes by comparing their digest changes.
+                    # The config digest of the previous build is stored at .oit/config_digest on distgit repo.
+                    dgr = meta.distgit_repo()
+                    path = Path(dgr.distgit_dir).joinpath(".oit", "config_digest")
+                    if not path.exists():  # alway request a buiild if config_digest hasn't been stored
+                        runtime.logger.info('%s config_digest does not exist; request a build', dgk)
+                        needs_rebuild = True
+                        reason = '.oit/config_digest does not exist'
```

Comment:
I think your suggestion of ignoring missing `config_digest` to avoid the initial mass rebuild is good. Can we just:
```suggestion
                        pass  # TODO: uncomment the following after a week of natural rebuilds
                        # runtime.logger.info('%s config_digest does not exist; request a build', dgk)
                        # needs_rebuild = True
                        # reason = '.oit/config_digest does not exist'
```
... and plan to uncomment it next week?